// @wordpress/block-editor/build/components/inserter/block-patterns-tab/pattern-category-preview-panel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PatternCategoryPreviewPanel = PatternCategoryPreviewPanel;
var _patternCategoryPreviews = require("./pattern-category-previews");
var _useZoomOut = require("../../../hooks/use-zoom-out");
var _jsxRuntime = require("react/jsx-runtime");
/**
 * Internal dependencies
 */

function PatternCategoryPreviewPanelInner({
  rootClientId,
  onInsert,
  onHover,
  category,
  showTitlesAsTooltip,
  patternFilter
}) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_patternCategoryPreviews.PatternCategoryPreviews, {
    rootClientId: rootClientId,
    onInsert: onInsert,
    onHover: onHover,
    category: category,
    showTitlesAsTooltip: showTitlesAsTooltip,
    patternFilter: patternFilter
  }, category.name);
}
function PatternCategoryPreviewPanelWithZoomOut(props) {
  (0, _useZoomOut.useZoomOut)();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(PatternCategoryPreviewPanelInner, {
    ...props
  });
}
function PatternCategoryPreviewPanel(props) {
  // When the pattern panel is showing, we want to use zoom out mode
  if (window.__experimentalEnableZoomedOutPatternsTab) {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)(PatternCategoryPreviewPanelWithZoomOut, {
      ...props
    });
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(PatternCategoryPreviewPanelInner, {
    ...props
  });
}
