// @wordpress/components/build/custom-select-control-v2/custom-select.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.CustomSelectContext = void 0;
var _element = require("@wordpress/element");
var _i18n = require("@wordpress/i18n");
var _ = require("..");
var Styled = _interopRequireWildcard(require("./styles"));
var _inputBase = _interopRequireDefault(require("../input-control/input-base"));
var _chevronDown = _interopRequireDefault(require("../select-control/chevron-down"));
var _jsxRuntime = require("react/jsx-runtime");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

const CustomSelectContext = exports.CustomSelectContext = (0, _element.createContext)(undefined);
function defaultRenderSelectedValue(value) {
  const isValueEmpty = Array.isArray(value) ? value.length === 0 : value === undefined || value === null;
  if (isValueEmpty) {
    return (0, _i18n.__)('Select an item');
  }
  if (Array.isArray(value)) {
    return value.length === 1 ? value[0] :
    // translators: %s: number of items selected (it will always be 2 or more items)
    (0, _i18n.sprintf)((0, _i18n.__)('%s items selected'), value.length);
  }
  return value;
}
const CustomSelectButton = ({
  renderSelectedValue,
  size = 'default',
  store,
  ...restProps
}) => {
  const {
    value: currentValue
  } = store.useState();
  const computedRenderSelectedValue = (0, _element.useMemo)(() => renderSelectedValue !== null && renderSelectedValue !== void 0 ? renderSelectedValue : defaultRenderSelectedValue, [renderSelectedValue]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(Styled.Select, {
    ...restProps,
    size: size,
    hasCustomRenderProp: !!renderSelectedValue,
    store: store
    // to match legacy behavior where using arrow keys
    // move selection rather than open the popover
    ,
    showOnKeyDown: false,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      children: computedRenderSelectedValue(currentValue)
    })
  });
};
function _CustomSelect(props) {
  const {
    children,
    hideLabelFromVision = false,
    label,
    size,
    store,
    ...restProps
  } = props;
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [hideLabelFromVision ?
    /*#__PURE__*/
    // TODO: Replace with BaseControl
    (0, _jsxRuntime.jsx)(_.VisuallyHidden, {
      as: "label",
      children: label
    }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(Styled.SelectLabel, {
      store: store,
      children: label
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)(_inputBase.default, {
      __next40pxDefaultSize: true,
      size: size,
      suffix: /*#__PURE__*/(0, _jsxRuntime.jsx)(_chevronDown.default, {}),
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(CustomSelectButton, {
        ...restProps,
        size: size,
        store: store
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(Styled.SelectPopover, {
        gutter: 12,
        store: store,
        sameWidth: true,
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(CustomSelectContext.Provider, {
          value: {
            store
          },
          children: children
        })
      })]
    })]
  });
}
var _default = exports.default = _CustomSelect;
