// @wordpress/block-editor/build/components/block-switcher/preview-block-popover.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PreviewBlockPopover;
var _i18n = require("@wordpress/i18n");
var _components = require("@wordpress/components");
var _blockPreview = _interopRequireDefault(require("../block-preview"));
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

function PreviewBlockPopover({
  blocks
}) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: "block-editor-block-switcher__popover__preview__parent",
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "block-editor-block-switcher__popover__preview__container",
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.Popover, {
        className: "block-editor-block-switcher__preview__popover",
        placement: "bottom-start",
        focusOnMount: false,
        children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
          className: "block-editor-block-switcher__preview",
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
            className: "block-editor-block-switcher__preview-title",
            children: (0, _i18n.__)('Preview')
          }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_blockPreview.default, {
            viewportWidth: 500,
            blocks: blocks
          })]
        })
      })
    })
  });
}
