// @wordpress/block-editor/build/components/inspector-controls-tabs/settings-tab-hint.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = InspectorControlsTabsHint;
var _components = require("@wordpress/components");
var _data = require("@wordpress/data");
var _dom = require("@wordpress/dom");
var _element = require("@wordpress/element");
var _i18n = require("@wordpress/i18n");
var _icons = require("@wordpress/icons");
var _preferences = require("@wordpress/preferences");
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

const PREFERENCE_NAME = 'isInspectorControlsTabsHintVisible';
function InspectorControlsTabsHint() {
  const isInspectorControlsTabsHintVisible = (0, _data.useSelect)(select => {
    var _select$get;
    return (_select$get = select(_preferences.store).get('core', PREFERENCE_NAME)) !== null && _select$get !== void 0 ? _select$get : true;
  }, []);
  const ref = (0, _element.useRef)();
  const {
    set: setPreference
  } = (0, _data.useDispatch)(_preferences.store);
  if (!isInspectorControlsTabsHintVisible) {
    return null;
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    ref: ref,
    className: "block-editor-inspector-controls-tabs__hint",
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "block-editor-inspector-controls-tabs__hint-content",
      children: (0, _i18n.__)("Looking for other block settings? They've moved to the styles tab.")
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.Button, {
      className: "block-editor-inspector-controls-tabs__hint-dismiss",
      icon: _icons.close,
      iconSize: "16",
      label: (0, _i18n.__)('Dismiss hint'),
      onClick: () => {
        // Retain focus when dismissing the element.
        const previousElement = _dom.focus.tabbable.findPrevious(ref.current);
        previousElement?.focus();
        setPreference('core', PREFERENCE_NAME, false);
      },
      showTooltip: false
    })]
  });
}
