// @wordpress/block-editor/build-module/components/inserter/block-patterns-tab/pattern-category-preview-panel.js
/**
 * Internal dependencies
 */
import { PatternCategoryPreviews } from './pattern-category-previews';
import { useZoomOut } from '../../../hooks/use-zoom-out';
import { jsx as _jsx } from "react/jsx-runtime";
function PatternCategoryPreviewPanelInner({
  rootClientId,
  onInsert,
  onHover,
  category,
  showTitlesAsTooltip,
  patternFilter
}) {
  return /*#__PURE__*/_jsx(PatternCategoryPreviews, {
    rootClientId: rootClientId,
    onInsert: onInsert,
    onHover: onHover,
    category: category,
    showTitlesAsTooltip: showTitlesAsTooltip,
    patternFilter: patternFilter
  }, category.name);
}
function PatternCategoryPreviewPanelWithZoomOut(props) {
  useZoomOut();
  return /*#__PURE__*/_jsx(PatternCategoryPreviewPanelInner, {
    ...props
  });
}
export function PatternCategoryPreviewPanel(props) {
  // When the pattern panel is showing, we want to use zoom out mode
  if (window.__experimentalEnableZoomedOutPatternsTab) {
    return /*#__PURE__*/_jsx(PatternCategoryPreviewPanelWithZoomOut, {
      ...props
    });
  }
  return /*#__PURE__*/_jsx(PatternCategoryPreviewPanelInner, {
    ...props
  });
}
