// @wordpress/block-editor/build/components/block-quick-navigation/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = BlockQuickNavigation;
var _data = require("@wordpress/data");
var _components = require("@wordpress/components");
var _blocks = require("@wordpress/blocks");
var _store = require("../../store");
var _blockIcon = _interopRequireDefault(require("../block-icon"));
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

function BlockQuickNavigation({
  clientIds
}) {
  if (!clientIds.length) {
    return null;
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.__experimentalVStack, {
    spacing: 1,
    children: clientIds.map(clientId => /*#__PURE__*/(0, _jsxRuntime.jsx)(BlockQuickNavigationItem, {
      clientId: clientId
    }, clientId))
  });
}
function BlockQuickNavigationItem({
  clientId
}) {
  const {
    name,
    icon,
    isSelected
  } = (0, _data.useSelect)(select => {
    const {
      getBlockName,
      getBlockAttributes,
      isBlockSelected,
      hasSelectedInnerBlock
    } = select(_store.store);
    const {
      getBlockType
    } = select(_blocks.store);
    const blockType = getBlockType(getBlockName(clientId));
    const attributes = getBlockAttributes(clientId);
    return {
      name: blockType && (0, _blocks.__experimentalGetBlockLabel)(blockType, attributes, 'list-view'),
      icon: blockType?.icon,
      isSelected: isBlockSelected(clientId) || hasSelectedInnerBlock(clientId, /* deep: */true)
    };
  }, [clientId]);
  const {
    selectBlock
  } = (0, _data.useDispatch)(_store.store);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.Button, {
    isPressed: isSelected,
    onClick: () => selectBlock(clientId),
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(_components.Flex, {
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_components.FlexItem, {
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_blockIcon.default, {
          icon: icon
        })
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.FlexBlock, {
        style: {
          textAlign: 'left'
        },
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_components.__experimentalTruncate, {
          children: name
        })
      })]
    })
  });
}
