// @wordpress/block-editor/build-module/components/block-switcher/preview-block-popover.js
/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Popover } from '@wordpress/components';

/**
 * Internal dependencies
 */
import BlockPreview from '../block-preview';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export default function PreviewBlockPopover({
  blocks
}) {
  return /*#__PURE__*/_jsx("div", {
    className: "block-editor-block-switcher__popover__preview__parent",
    children: /*#__PURE__*/_jsx("div", {
      className: "block-editor-block-switcher__popover__preview__container",
      children: /*#__PURE__*/_jsx(Popover, {
        className: "block-editor-block-switcher__preview__popover",
        placement: "bottom-start",
        focusOnMount: false,
        children: /*#__PURE__*/_jsxs("div", {
          className: "block-editor-block-switcher__preview",
          children: [/*#__PURE__*/_jsx("div", {
            className: "block-editor-block-switcher__preview-title",
            children: __('Preview')
          }), /*#__PURE__*/_jsx(BlockPreview, {
            viewportWidth: 500,
            blocks: blocks
          })]
        })
      })
    })
  });
}
